### ISSUE TITLE
Dialog forms doesn't work properly

### ISSUE THREAD

**Issue by robhunt3r** [sonata-project/SonataDoctrinePhpcrAdminBundle#309]
I was trying to use inline forms in order to add media/gallery/stuff to other stuff, after the PRs I did for MediaBundle I made it working, but problem was now that when I clicked on LIST/CREATE/DELETE buttons I've found some problems:

Everything is opening with jQueryUI dialogs instead of Bootstrap modals so it looked ugly.
List button worked perfectly.
Create button didn't work, undefined related errors with event.preventDefault(); when clicked.
Delete button worked.

So after research, I've got it working, editing some stuff in comparison with ORMAdminBundle.
**NOTE**: This was done with Many association, not One, but I think problem will be the same

`:CRUD:edit_phpcr_many_association_script.html.twig` 
After editing some lines, I ended with a CREATE dialog, but when clicking on SUBMIT it was causing a redirection to it's original page, so useless. Finally I ended duplicating ORM code, for the same file, this was causing a dark overlay over all page without contents.

The problem was easy to find: There were NOT a modal div (as I said it was using jQueryUI Dialog) this is setted inside `FORM:form_admin_fields.html.twig` replacing this

``` html


        <div class="container sonata-ba-modal sonata-ba-modal-edit-one-to-one" style="display: none" id="field_dialog_{{ id }}">

        </div>
```

with this

``` html


<div class="modal fade" id="field_dialog_{{ id }}" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true">
    <div class="modal-dialog modal-lg">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
                <h4 class="modal-title"></h4>
            </div>
            <div class="modal-body">
            </div>
        </div>
    </div>
</div>
```

This code is inside `SonataOrmAdminBundle:CRUD:edit_moda.html.twig`

With this changes everything is working perfectly.

Sorry for this bad explaination, I'm in a hurry while writting. I hope that you understand the problem and the solution I've used.

**_TL;DR / WTF are you saying**_
I think we can use the same Javascript + twig code for ORM and PHPCR admin bundles, because they work in both bundles, at least under a Many relation which is what I'm using.

Maybe we can create common JS+TWIG files inside AdminBundle as AdminBundle requires at least one DB bunle and make ORM/PHPCR/MONGODB use it.

In order to reproduce this problem just create the simplest web with mediabundle with my PR (see my profile) using PHPCR and try to add media to a Gallery changing what I said here (JS files and modal DIV)

**Comment by lsmith77** [comment]
ping

**Comment by ildikototh** [comment]
I also have a problem with the dialog form. 
in the  edit_phpcr_many_association_script.html.twig line 346 i think it's wrongly call the add functionality, therefore that doesn't open a  modal, but a  new window, also throws event.preventDefault(); error in the browser. 
The original line in the  function start_field_dialog_form_add_{{ id }}(link)  code is this:
```field_dialog_form_add_{{ id }}(link);```

I just recommend to fix to: 

```
// add the jQuery event to the a element
        jQuery(link)
            .click(field_dialog_form_add_{{ id }})
            .trigger('click')
        ;
```

**Comment by dbu** [comment]
this is being worked on: https://github.com/sonata-project/SonataAdminBundle/pull/3971

glad if you can help test that PR
